### DISCUSSION TITLE
How to `join` with an empty string?

### DISCUSSION THREAD

**Discussion by bravoecho** [TomWright/dasel#321] [Q&A] [answered]
I would like to use the `join` function with an empty string as the separator, but I cannot find a way. If I leave the first argument empty, as in `join(,field1,field2)`, the behaviour becomes unpredictable depending on the type of the other arguments (e.g. literal `string` or field names), but more often than not it will drop `field1`.

As a very contrived example, if I run:

```
 echo '{ "filename": "a.txt", "size": "1MB" }
{ "filename": "b.txt", "size": "0.5MB" }
{ "filename": "c.txt", "size": "1.5MB" }
{ "filename": "d.txt", "size": "0.3MB" }' \
> | dasel -r json -w csv 'all().mapOf(file_size,size,file_path,this().join(,string(/some/dir/),filename)).merge()'
```

...then I get...

```
file_path,file_size
a.txt,1MB
b.txt,0.5MB
c.txt,1.5MB
d.txt,0.3MB
```

...whereas I would expect...

```
file_path,file_size
/some/dir/a.txt,1MB
/some/dir/b.txt,0.5MB
/some/dir/c.txt,1.5MB
/some/dir/d.txt,0.3MB
```

I have tried a combination of escaping and so on, but just can't get it to work. 

Thanks if you can help! 🙂

**Comment by TomWright** [comment, marked as answer]
You should be able to use `string()` with no args to represent an empy string, however I wouldn't join with an empty string in this case, rather join with `/`.

E.g. This worked for me:
```bash
$ echo '{ "filename": "a.txt", "size": "1MB" }
{ "filename": "b.txt", "size": "0.5MB" }
{ "filename": "c.txt", "size": "1.5MB" }
{ "filename": "d.txt", "size": "0.3MB" }' | dasel -r json -w csv 'all().mapOf(file_path,join(/,string(/some/dir),filename),file_size,size).merge()'
file_path,file_size
/some/dir/a.txt,1MB
/some/dir/b.txt,0.5MB
/some/dir/c.txt,1.5MB
/some/dir/d.txt,0.3MB
```

**Reply by TomWright** [reply]
As a side note, I just discovered the separator for `join` isn't parsed as a selector, so it should infact be `join(,x,y)` to join `x` and `y` without any spaces.

I can do further testing on this in the future but the answer above should solve your problem.
